docs/src/main/sphinx/connector/mariadb.rst

```diff
@@ -71,41 +71,199 @@ that catalog name instead of ``mariadb`` in the above examples.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps MariaDB types to the corresponding Trino types according
+to the following table:
+
+.. list-table:: MariaDB type to Trino type mapping
+  :widths: 30, 30, 50
+  :header-rows: 1
+
+  * - MariaDB type
+    - Trino type
+    - Notes
+  * - ``boolean``
+    - ``tinyint``
+    -
+  * - ``tinyint``
+    - ``tinyint``
+    -
+  * - ``smallint``
+    - ``smallint``
+    -
+  * - ``int``
+    - ``integer``
+    -
+  * - ``bigint``
+    - ``bigint``
+    -
+  * - ``tinyint unsigned``
+    - ``smallint``
+    -
+  * - ``smallint unsigned``
+    - ``integer``
+    -
+  * - ``mediumint unsigned``
+    - ``integer``
+    -
+  * - ``integer unsigned``
+    - ``bigint``
+    -
+  * - ``bigint unsigned``
+    - ``decimal(20,0)``
+    -
+  * - ``float``
+    - ``real``
+    -
+  * - ``double``
+    - ``double``
+    -
+  * - ``decimal(p,s)``
+    - ``decimal(p,s)``
+    -
+  * - ``char(n)``
+    - ``char(n)``
+    -
+  * - ``tinytext``
+    - ``varchar(255)``
+    -
+  * - ``text``
+    - ``varchar(65535)``
+    -
+  * - ``mediumtext``
+    - ``varchar(16777215)``
+    -
+  * - ``longtext``
+    - ``varchar``
+    -
+  * - ``varchar(n)``
+    - ``varchar(n)``
+    -
+  * - ``tinyblob``
+    - ``varbinary``
+    -
+  * - ``blob``
+    - ``varbinary``
+    -
+  * - ``longblob``
+    - ``varbinary``
+    -
+  * - ``varbinary(n)``
+    - ``varbinary``
+    -
+  * - ``date``
+    - ``date``
+    -
+  * - ``time(n)``
+    - ``time(n)``
+    -
+  * - ``timestamp(n)``
+    - ``timestamp(n)``
+    - MariaDB stores the current timestamp by default. Please enable
+      `explicit_defaults_for_timestamp <https://mariadb.com/docs/reference/mdb/system-variables/explicit_defaults_for_timestamp/>`_
+      to avoid implicit default values.
+
+Trino type mapping to MariaDB type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps Trino types to the corresponding MariaDB types according
+to the following table:
+
+.. list-table:: Trino type mapping to MariaDB type mapping
+  :widths: 30, 25, 50
+  :header-rows: 1
+
+  * - MariaDB type
+    - Trino type
+    - Notes
+  * - ``tinyint``
+    - ``boolean``
+    -
+  * - ``tinyint``
+    - ``tinyint``
+    -
+  * - ``smallint``
+    - ``smallint``
+    -
+  * - ``integer``
+    - ``int``
+    -
+  * - ``bigint``
+    - ``bigint``
+    -
+  * - ``smallint``
+    - ``tinyint unsigned``
+    -
+  * - ``integer``
+    - ``smallint unsigned``
+    -
+  * - ``integer``
+    - ``mediumint unsigned``
+    -
+  * - ``bigint``
+    - ``integer unsigned``
+    -
+  * - ``decimal(20,0)``
+    - ``bigint unsigned``
+    -
+  * - ``real``
+    - ``float``
+    -
+  * - ``double``
+    - ``double``
+    -
+  * - ``decimal(p,s)``
+    - ``decimal(p,s)``
+    -
+  * - ``char(n)``
+    - ``char(n)``
+    -
+  * - ``varchar(255)``
+    - ``tinytext``
+    -
+  * - ``varchar(65535)``
+    - ``text``
+    -
+  * - ``varchar(16777215)``
+    - ``mediumtext``
```

Comment:
These mapping looks confusing. The connector maps to `tinytext` when the length <= 255, not exact 255.

Reply:
These are from the original mapping. @ebyhr do you want me to add <= to all those entries?

Reply:
The third column of the table is used for notes, so you can add a note to each of these respective entries like:

* Maps on `VARCHAR` of length 255 or less.
* Maps on `VARCHAR` of length between 256 and 65535, inclusive.
* Maps on `VARCHAR` of length between 65536 and 16777215, inclusive.

Reply:
Done only in Trino --> MariaDB table